csm/common/ha/cluster_management/operations.py

```diff
@@ -0,0 +1,140 @@
+        error_messages = []
+        for each_key in errors.keys():
+            if errors[each_key][0] == const.UNKNOWN_FIELD_ERR_MSG:
+                error_messages.append(f"{each_key}: {const.ARG_NOT_SUPPORTED_ERR_MSG}")
+            else:
+                error_messages.append(f"{each_key}: {''.join(errors[each_key])}")
+
+        if len(error_messages) > 0:
+            raise InvalidRequest('\n'.join(error_messages))
+
+
+class ClusterStartOperation(Operation):
+
+    def validate_arguments(self, **kwargs):
+        pass
+
+    def execute(self, cluster_manager, **kwargs):
+        pass
+
+
+class ClusterStopOperation(Operation):
+
+    def validate_arguments(self, **kwargs):
+        pass
+
+    def execute(self, cluster_manager, **kwargs):
+        pass
+
+
+class NodeStartOperation(Operation):
+
+    def validate_arguments(self, **kwargs):
+        pass
+
+    def execute(self, cluster_manager, **kwargs):
+        pass
+
+
+class NodeStopOperation(Operation):
+
+    def validate_arguments(self, **kwargs):
+        fields_to_validate = {
+            const.ARG_RESOURCE_ID: fields.Str(required=True,
+                                    validate=self.not_blank_validator),
+            const.ARG_FORCE: fields.Bool()
+        }
+        ValidatorSchema = Schema.from_dict(fields_to_validate)
+        errors = ValidatorSchema().validate(kwargs)
+        self.parse_errors(errors)
+
+    def execute(self, cluster_manager, **kwargs):
+        node_id = kwargs.get(const.ARG_RESOURCE_ID, "")
+        args = {
+            const.ARG_CHECK_CLUSTER: not kwargs.get(const.ARG_FORCE, False)
+        }
+        Log.debug(f"NodeStopOperation on node with id {node_id} and args: {args}")
+        # try:
+        #     operation_result = cluster_manager.node_controller.stop(node_id, args)
+        #     Log.debug(f"NodePoweroffOperation result: {operation_result}")
+        # except Exception as e:
+        #     err_msg = f"{const.CLUSTER_OPERATIONS_ERR_MSG} : {e}"
+        #     Log.error(err_msg)
+
+
+class NodePoweroffOperation(Operation):
+
+    def validate_arguments(self, **kwargs):
+        fields_to_validate = {
+            const.ARG_RESOURCE_ID: fields.Str(required=True,
+                                    validate=self.not_blank_validator),
+            const.ARG_FORCE: fields.Bool(),
+            const.ARG_STORAGE_OFF: fields.Bool()
+        }
+        ValidatorSchema = Schema.from_dict(fields_to_validate)
+        errors = ValidatorSchema().validate(kwargs)
+        self.parse_errors(errors)
+
+    def execute(self, cluster_manager, **kwargs):
+        node_id = kwargs.get(const.ARG_RESOURCE_ID, "")
+        args = {
+            const.ARG_CHECK_CLUSTER: not kwargs.get(const.ARG_FORCE, False),
+            const.ARG_POWER_OFF: True,
+            const.ARG_STORAGE_OFF: kwargs.get(const.ARG_STORAGE_OFF, False)
+        }
+        Log.debug(f"NodePoweroffOperation on node with id {node_id} and args: {args}")
+        # try:
```

Comment:
Remove the commented lines. We will merge along with HA

Reply:
done